src/v/pandaproxy/schema_registry/store.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#pragma once
+
+#include "likely.h"
+#include "outcome.h"
+#include "pandaproxy/schema_registry/error.h"
+#include "pandaproxy/schema_registry/types.h"
+
+#include <absl/container/btree_map.h>
+#include <absl/container/node_hash_map.h>
+#include <absl/hash/hash.h>
+
+#include <functional>
+#include <iterator>
+#include <system_error>
+
+namespace pandaproxy::schema_registry {
+
+class store {
+public:
+    ///\brief Insert a schema for a given subject.
+    ///
+    /// If the schema is not registered, register it.
+    /// If the subject does not have this schema at any version, register a new
+    /// version.
+    ///
+    /// return the schema_version and schema_id, and whether it's new.
+    std::pair<subject_version_id, bool>
+    insert(subject sub, schema_definition def, schema_type type) {
+        auto id = insert_schema(std::move(def), type).first;
+        auto [version, inserted] = insert_subject(std::move(sub), id);
+        return {{version, id}, inserted};
+    }
+
+private:
+    std::pair<schema_id, bool>
+    insert_schema(schema_definition def, schema_type type) {
+        const auto s_it = std::find_if(
+          _schemas.begin(), _schemas.end(), [&](const auto& s) {
+              const auto& entry{s.second};
+              return type == entry.type && def == entry.definition;
+          });
+        if (s_it != _schemas.end()) {
+            return {s_it->first, false};
+        }
+
+        const auto id = _schemas.empty() ? schema_id{1}
```

Comment:
this code is repeated. def extract it to a method. Also, is this in the hotpath? should we put it in the .cc file instead. 

I think this could be a simple 'inline' method on a .cc file.

Reply:
ah NVM!!! sorry the other is for versions, not for schemas...